vizro-core/src/vizro/models/_components/form/date_picker.py

```diff
@@ -37,11 +36,13 @@ class DatePicker(VizroBaseModel):
 
     """
 
-    type: Literal["date_picker"] = "date_picker"
+    type: Literal["default", "range", "multiple"] = "default"
     min: Optional[date] = Field(None, description="Start date for date picker.")
     max: Optional[date] = Field(None, description="End date for date picker.")
     value: Optional[Union[list[date], date]] = Field(None, description="Default date for date picker")
     title: str = Field("", description="Title to be displayed.")
+
+    # Could probably delete the `range` arg, but keeping it makes it backwards compatible
```

Comment:
Good point! The `range` argument is definitely redundant if we ever want to introduce the `type` argument of the new `dmc.DatePickerInput` as well. Options would then include: "default", "range", "multiple" in which case this boolean argument doesn't make much sense. 

Let's leave it in for now and discuss when @antonymilne is back 👍

Reply:
Ah, perfect! I didn't know he commented on this already. Let's leave it as is then, and @AnnMarieW you could delete this line then: `date_range_picker_kwargs = {"allowSingleDateInRange": True} if self.range else {}`

I don't think it's needed anymore since that is controlled by the `type` argument now instead of `allowSingleDateInRange`

Reply:
We should always include `allowSingleDateInRange=True,` in the `dmc.DatePickerInput` configuration. It will enable that a single date can be selected when `range=True` (so `type="range"`), and will have no effect when `range=False` (so `type="default"`).

Reply:
Ah I see, but then let's specify it directly as ` allowSingleDateInRange=True` instead of adding that line 👍

Reply:
I'm a little confused by this - not sure what props to update :thinking:

Reply:
All that is needed is to delete the following line:
`        date_range_picker_kwargs = {"allowSingleDateInRange": True} if self.range else {}`

and insert `allowSingleDateInRange: True` directly to the `dmc.DatePickerInput` configuration like in this [suggestion comment](https://github.com/mckinsey/vizro/pull/924/files#r1888005520).